### PR TITLE
chore: bump depedencies

### PR DESCRIPTION
**Pull request by ndelanou**
Upgrade project dependencies, blocking other packages that also depend on it to upgrade (e.g. `melos` and `dart_code_metrics` via `pub_updater`)

Also bump the SDK version to <4.0.0

All test passed.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -4,13 +4,13 @@ description: A pluggable, mockable process invocation abstraction for Dart.
 homepage: https://github.com/google/process.dart
 
 environment:
-  sdk: '>=2.14.0-0 <3.0.0'
+  sdk: '>=2.14.0-0 <4.0.0'
 
 dependencies:
-  file: '^6.0.0'
+  file: '^7.0.0'
```
Comment (Gustl22) [inline pubspec.yaml]:
```suggestion
  file: '>=6.0.0 <8.0.0'
```

You may can use `file: '>=6.0.0 <8.0.0'` as there weren't any changes which had to be made and it provides much broader compatibility.

File: pubspec.yaml [hunk 2]
```diff
@@ -4,13 +4,13 @@ description: A pluggable, mockable process invocation abstraction for Dart.
...
   path: ^1.8.0
   platform: '^3.0.0'
 
 dev_dependencies:
-  lints: ^1.0.1
+  lints: ^2.1.0
```
Comment (Gustl22) [inline pubspec.yaml]:
```suggestion
  lints: '>=1.0.1 <3.0.0'
```
Same for lints: `lints: '>=1.0.1 <3.0.0'`

**Review by Gustl22 — CHANGES_REQUESTED** [review]

**Comment by Gustl22** [comment]
This affects all Linux implementations, e.g. in [`path_provider`](https://github.com/flutter/packages/blob/main/packages/path_provider/path_provider_linux/pubspec.yaml), [xdg_directories](https://github.com/flutter/packages/blob/2d573fe913ebddeacb4ea869461b3b419aa48b5b/packages/xdg_directories/pubspec.yaml#L16), as one can't upgrade to `file >= 7.0.0` @jonahwilliams @tvolkert

**Review by SirBarksALot — APPROVED** [review]

**Comment by bernaferrari** [comment]
cc @jakemac53 could you help this land?

**Comment by jakemac53** [comment]
I am not really affiliated with this package at all, it looks like @gspencergoog or @jonahwilliams have done some work though.

We do need a pubspec/changelog update for a new version (can just be a patch release, so `5.0.1`).

**Comment by gspencergoog** [comment]
This looks fine to me, but it does need to have the CHANGELOG updated and the version bumped, as @jakemac53 said.

**Pull request closed without merging**